command/vsm_stats.go

```diff
@@ -0,0 +1,133 @@
+package command
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/rancher/go-rancher/client"
+)
+
+type Stats struct {
+	client.Resource
+	ReplicaCounter  int64 `json:"replicacounter"`
+	RevisionCounter int64 `json:"revisioncounter"`
+}
+
+type VsmStatsCommand struct {
+	Meta
+	address     string
+	host        string
+	length      int
+	replica_ips string
+}
+
+type ReplicaClient struct {
+	address    string
+	syncAgent  string
+	host       string
+	httpClient *http.Client
+}
+
+func (c *VsmStatsCommand) Help() string {
+	helpText := `
+	Usage: maya vsm-stats [options] <replica-ip:port> 
+
+  Display stats information about VSM.
+
+  Stats Options:
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (c *VsmStatsCommand) Synopsis() string {
+	return "Display information about Vsm(s)"
+}
+
+func (c *VsmStatsCommand) Run(args []string) int {
+
+	var stats Stats
+	flags := c.Meta.FlagSet("vsm-stats", FlagSetClient)
+	flags.Usage = func() { c.Ui.Output(c.Help()) }
+	flags.StringVar(&c.replica_ips, "replica-ips", "", "")
+
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+	args = flags.Args()
+	if len(args) < 1 {
+		c.Ui.Error(c.Help())
+		return 1
+	}
+	fmt.Printf("%2s %15s\n", "RevisionCounter", "ReplicaCounter")
+	for counter := 0; counter <= 100; counter++ {
+		time.Sleep(2 * time.Second)
+		err := GetStatus(args[0], &stats)
+		if err != nil {
+			fmt.Println("\nERROR:", err)
+			return -1
+		}
+		fmt.Printf("\r    %-17d %d", stats.RevisionCounter, stats.ReplicaCounter)
+	}
+	return 0
+}
+
+func NewReplicaClient(address string) (*ReplicaClient, error) {
```

Comment:
Lot of technical stuff are programmed here in this func
I would have provided code level comments that helps one understand the need of each.